src/AspNetCoreModuleV2/AspNetCore/src/aspnetcore_shim_config.cpp

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "precomp.hxx"
+
+ASPNETCORE_SHIM_CONFIG::~ASPNETCORE_SHIM_CONFIG()
+{
+    if (m_ppStrArguments != NULL)
+    {
+        delete[] m_ppStrArguments;
+        m_ppStrArguments = NULL;
+    }
+}
+
+VOID
+ASPNETCORE_SHIM_CONFIG::ReferenceConfiguration(
+    VOID
+) const
+{
+    InterlockedIncrement(&m_cRefs);
+}
+
+VOID
+ASPNETCORE_SHIM_CONFIG::DereferenceConfiguration(
+    VOID
+) const
+{
+    DBG_ASSERT(m_cRefs != 0);
+    LONG cRefs = 0;
+    if ((cRefs = InterlockedDecrement(&m_cRefs)) == 0)
+    {
+        delete this;
+    }
+}
+
+HRESULT
+ASPNETCORE_SHIM_CONFIG::GetConfig(
+    _In_  IHttpServer             *pHttpServer,
+    _In_  HTTP_MODULE_ID           pModuleId,
+    _In_  IHttpApplication        *pHttpApplication,
+    _In_  HANDLE                   hEventLog,
+    _Out_ std::wstring            *exeLocation,
+    _Out_ ASPNETCORE_SHIM_CONFIG **ppAspNetCoreShimConfig
+)
+{
+    HRESULT                 hr = S_OK;
+    ASPNETCORE_SHIM_CONFIG *pAspNetCoreShimConfig = NULL;
+    STRU                    struHostFxrDllLocation;
+    STRU                    struExeAbsolutePath;
+    BSTR*                   pwzArgv;
+    DWORD                   dwArgCount;
+    if (ppAspNetCoreShimConfig == NULL)
+    {
+        hr = E_INVALIDARG;
+        goto Finished;
+    }
+
+    *ppAspNetCoreShimConfig = NULL;
+
+    // potential bug if user sepcific config at virtual dir level
+    pAspNetCoreShimConfig = (ASPNETCORE_SHIM_CONFIG*)
+        pHttpApplication->GetModuleContextContainer()->GetModuleContext(pModuleId);
+
+    if (pAspNetCoreShimConfig != NULL)
+    {
+        *ppAspNetCoreShimConfig = pAspNetCoreShimConfig;
+        pAspNetCoreShimConfig = NULL;
+        goto Finished;
+    }
+
+    pAspNetCoreShimConfig = new ASPNETCORE_SHIM_CONFIG;
+    if (pAspNetCoreShimConfig == NULL)
+    {
+        hr = E_OUTOFMEMORY;
+        goto Finished;
+    }
+
+    hr = pAspNetCoreShimConfig->Populate(pHttpServer, pHttpApplication);
+    if (FAILED(hr))
+    {
+        goto Finished;
+    }
+
+    // Modify Inprocess specific configuration here. 
+    if (pAspNetCoreShimConfig->QueryHostingModel() == APP_HOSTING_MODEL::HOSTING_IN_PROCESS)
```

Comment:
this piece of code seems wired here as it changes the configuration. It either should be part of ShimConfig::Populate or other method where use the HostFxr

Reply:
I'd prefer to not store state on the config at all from hostfxr. I can do this in a separate PR though.